### PR TITLE
Fix ESP32 compiler errors

### PR DESCRIPTION
**Pull request by caternuson**
For #19.

* Move static prototypes from .h to .cpp
* Remove numerous unused variables

### CODE REVIEW

File: src/Adafruit_BNO08x.cpp [hunk 1]
```diff
@@ -288,7 +312,7 @@ static int i2chal_read(sh2_Hal_t *self, uint8_t *pBuffer, unsigned len,
                        uint32_t *t_us) {
   // Serial.println("I2C HAL read");
 
-  uint8_t *pBufferOrig = pBuffer;
+  // uint8_t *pBufferOrig = pBuffer;
```
Comment (dhalbert) [inline src/Adafruit_BNO08x.cpp:315]: Just delete this instead?

File: src/Adafruit_BNO08x.h [hunk 2]
```diff
@@ -64,28 +64,28 @@ class Adafruit_BNO08x {
...
+// static int spihal_write(sh2_Hal_t *self, uint8_t *pBuffer, unsigned len);
+// static int spihal_read(sh2_Hal_t *self, uint8_t *pBuffer, unsigned len,
+//                        uint32_t *t_us);
+// static void spihal_close(sh2_Hal_t *self);
+// static int spihal_open(sh2_Hal_t *self);
+
+// static uint32_t hal_getTimeUs(sh2_Hal_t *self);
+// static void hal_callback(void *cookie, sh2_AsyncEvent_t *pEvent);
+// static void sensorHandler(void *cookie, sh2_SensorEvent_t *pEvent);
+// static void hal_hardwareReset(void);
```
Comment (dhalbert) [inline src/Adafruit_BNO08x.h:89]: Did you mean just to delete these lines (after testing them commented out)?

**Review by dhalbert — CHANGES_REQUESTED** [review]
Thanks - wondering about commented-out stuff.

File: src/Adafruit_BNO08x.cpp (hunk 1 above)
Comment (caternuson) [reply src/Adafruit_BNO08x.cpp:315]: this one is potentially used later in commented out debug prints, so needs to remain as a commented out line.

File: src/Adafruit_BNO08x.h (hunk 2 above)
Comment (caternuson) [reply src/Adafruit_BNO08x.h]: yes. thanks. these weren't meant to be deleted. fix was to just move these to the .cpp file.

**Review by dhalbert — APPROVED** [review]
lgtm - thanks!

**Pull request merged**